ee/vellum_ee/workflows/display/tests/workflow_serialization/generic_nodes/test_attributes_serialization.py

```diff
@@ -670,3 +670,62 @@ class GenericNodeWithPydantic(BaseNode):
     assert any(
         entry["key"] == "node_ref" and entry["value"]["type"] == "NODE_OUTPUT" for entry in attr_value["entries"]
     )
+
+
+def test_serialize_workflow__dataclass_output_with_preserved_type_metadata():
+    """Test that dataclass outputs preserve type metadata during workflow serialization."""
+
+    @dataclass
+    class CustomDataclassOutput:
+        result: str
+        count: int
+
+    class NodeWithDataclassOutput(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            custom_output: CustomDataclassOutput
+
+    class Workflow(BaseWorkflow):
+        graph = NodeWithDataclassOutput
+
+    from vellum_ee.workflows.display.workflows.get_vellum_workflow_display_class import get_workflow_display
+
+    workflow_display = get_workflow_display(workflow_class=Workflow)
+
+    serialized_workflow = workflow_display.serialize()
+
+    # THEN the workflow should be serialized with dataclass type metadata preserved
+    assert "workflow_raw_data" in serialized_workflow
+    workflow_raw_data = serialized_workflow["workflow_raw_data"]
+
+    nodes = workflow_raw_data["nodes"]
+    dataclass_node = None
+    for node in nodes:
+        definition = node.get("definition")
+        if definition and definition.get("name") == "NodeWithDataclassOutput":
+            dataclass_node = node
+            break
+
+    assert dataclass_node is not None, "Could not find NodeWithDataclassOutput in serialized nodes"
+
+    outputs = dataclass_node.get("outputs", [])
+    custom_output = None
+    for output in outputs:
+        if output.get("name") == "custom_output":
+            custom_output = output
+            break
+
+    assert custom_output is not None, "Could not find custom_output in node outputs"
+    assert custom_output["type"] == "JSON"
+
+    assert "type_definitions" in workflow_raw_data
+    type_definitions = workflow_raw_data["type_definitions"]
+    assert len(type_definitions) > 0
+
+    if custom_output.get("value") is not None:
+        assert custom_output["value"]["type"] == "TYPE_REFERENCE"
+        assert "type_definition_id" in custom_output["value"]
+
+        type_def_id = custom_output["value"]["type_definition_id"]
+        matching_type_def = next((td for td in type_definitions if td["id"] == type_def_id), None)
+        assert matching_type_def is not None
```

Comment:
improve this assertion to create an equality between `matching_type_def` and the expected dict